### PR TITLE
Deprecation of Props

### PR DESCRIPTION
**Pull request by thestephenmarshall**
This PR introduces the ability to deprecate kit props. [Documentation which describes how to use this new feature has also been added to the wiki](https://github.com/powerhome/playbook/wiki/Deprecation-of-Props) based off of these changes.

Screen Shot 2020-11-09 at 2 07 54 PM

Screen Shot 2020-11-09 at 2 08 15 PM

#### Breaking Changes

No

#### Runway Ticket URL

https://nitro.powerhrg.com/runway/teams/263/backlog/sprints/active#open-backlog-item-46592

#### How to test this

This cannot be tested currently. 

#### Checklist:

- [x] **LABELS** Add a label: `enhancement`, `bug`, `improvement`, `new kit`, `deprecated`, or `breaking`. See [Changelog & Labels](https://github.com/powerhome/playbook/wiki/Changelog-&-Labels) for details.
- [x] **URGENCY** Please select a release milestone
- [x] **DEPLOY** Please add the `Milano` label when you are ready for a review.
- [x] **SCREENSHOT** Please add a screen shot or two.
- [x] **SPECS** Please cover your changes with specs.
- [x] **READ DOCS** Please make sure you have read and understand the [Playbook Release Process](https://github.com/powerhome/playbook/wiki/Playbook-Releases)

*The normal release cut off deadline is 3p EDT each week. Please reach out to the release team if you have an urgent request or need a release off cycle.*

### CODE REVIEW

**Review by kre8sions — APPROVED** [review]

**Review by jasoncypret — APPROVED** [review]

**Comment by jasoncypret** [comment]
LGTM....waiting for the deploy

File: playbook/app/pb_kits/playbook/props/base.rb [hunk 1]
```diff
@@ -19,6 +20,8 @@ def value(value)
       end
 
       def validate!(input_value)
+        warn("#{kit} Kit: The prop '#{name}' is deprecated and will be removed a future release!") if deprecated #TODO: add some color for pop
```
Comment (garettarrowood) [inline playbook/app/pb_kits/playbook/props/base.rb:23]:
```suggestion
        warn("#{kit} Kit: The prop '#{name}' is deprecated and will be removed in a future release!") if deprecated #TODO: add some color for pop
```

File: playbook/app/pb_kits/playbook/utilities/globalProps.js [hunk 2]
```diff
@@ -44,3 +44,10 @@ export const globalProps = (props, defaultProps = {}) => {
   const allProps = { ...props, ...defaultProps }
   return spacingProps(allProps) + darkProps(allProps)
 }
+
+export const deprecatedProps = (kit, props = {}) => {
+  if (process.env.NODE_ENV === 'development') {
+    /* eslint no-console: ["error", { allow: ["warn", "error"] }] */
+    props.forEach((propName) => console.warn(`${kit} Kit: The prop '${propName}' is deprecated and will be removed a future release!`))
```
Comment (garettarrowood) [inline playbook/app/pb_kits/playbook/utilities/globalProps.js:51]:
```suggestion
    props.forEach((propName) => console.warn(`${kit} Kit: The prop '${propName}' is deprecated and will be removed in a future release!`))
```

**Review by garettarrowood — APPROVED** [review]
Other than adding a word, this LGTM!

**Pull request merged**